cilium-cli/install/aws.go

```diff
@@ -3,9 +3,99 @@
 
 package install
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+)
+
 const (
 	AwsNodeDaemonSetName              = "aws-node"
 	AwsNodeDaemonSetNamespace         = "kube-system"
 	AwsNodeDaemonSetNodeSelectorKey   = "io.cilium/aws-node-enabled"
 	AwsNodeDaemonSetNodeSelectorValue = "true"
 )
+
+const (
+	AwsNodeImageFamilyAmazonLinux2    = "AmazonLinux2"
+	AwsNodeImageFamilyAmazonLinux2023 = "AmazonLinux2023"
+	AwsNodeImageFamilyBottlerocket    = "Bottlerocket"
+	AwsNodeImageFamilyCustom          = "Custom"
+	AwsNodeImageFamilyUbuntu          = "Ubuntu"
+	AwsNodeImageFamilyWindows         = "Windows"
+)
+
+type awsClusterInfo struct {
+	ImageID string `json:"ImageID"`
+}
+
+func (k *K8sInstaller) awsRetrieveNodeImageFamily() error {
+	// setting default fallback value
+	k.params.AWS.AwsNodeImageFamily = AwsNodeImageFamilyCustom
+
+	bytes, err := k.eksctlExec("get", "nodegroup", "--cluster", k.client.ClusterName())
+	if err != nil {
+		k.Log("❌ Could not detect AWS node image family, defaulted to fallback value: %s", k.params.AWS.AwsNodeImageFamily)
+		return err
+	}
+
+	clusterInfo := awsClusterInfo{}
```

Comment:
I think `eksctl get nodegroup` command will return output as JSON array, isn't it?
If yes then `json.Unmarshal` into a struct will fail.